### PR TITLE
Add spanner handler

### PR DESCRIPTION
**Pull request by castaneai**

### CODE REVIEW

File: datasource/handler/sql.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func (h *SQLHandler) GetSchemas() (*[]Schema, error) {
 	defer sqlRows.Close()
 
 	// scan results
-	schemaMap := make(map[string]Schema)
+	schemaMap := make(map[string]*Schema)
```
Comment (Mitu217) [inline datasource/handler/sql.go]: I want to this
```
var schemaMap map[string]*Schema
```

File: datasource/handler/spanner.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+// GetSchemas is get all schemas method
+func (h *SpannerHandler) GetSchemas() ([]*Schema, error) {
+	ctx := context.Background()
+	stmt := spanner.NewStatement("SELECT TABLE_NAME, COLUMN_NAME, ORDINAL_POSITION, SPANNER_TYPE, IS_NULLABLE FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = ''")
+	iter := h.spannerClient.Single().Query(ctx, stmt)
+	defer iter.Stop()
+
+	// scan results
+	schemaMap := make(map[string]*Schema)
```
Comment (Mitu217) [inline datasource/handler/spanner.go]: I want to this
```
var schemaMap map[string]*Schema
```

File: datasource/handler/spanner.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"cloud.google.com/go/spanner"
+	"google.golang.org/api/iterator"
+)
+
+// SpreadsheetHandler is handler struct of csv
```
Comment (Mitu217) [inline datasource/handler/spanner.go]: Fix comment

File: datasource/handler/spanner.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+type SpannerHandler struct {
+	DSN           string `json:"dsn"`
+	spannerClient *spanner.Client
+}
+
+// NewSpreadsheetHandler is create SpreadsheetHandler instance method
```
Comment (Mitu217) [inline datasource/handler/spanner.go]: Fix comment

File: datasource/handler/spanner.go [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+					return nil, err
+				}
+				isPrimaryKey = true
+			}
+
+			if isPrimaryKey {
+				schema.PrimaryKey = columnName
+				schema.primaryKeyIndex = int(ordinalPosition) - 1
+			}
+		}
```
Comment (Mitu217) [inline datasource/handler/spanner.go]: I want to make get primary key action more simply...
Is spanner support COLUMN_KEY like mysql?

File: datasource/handler/csv.go [hunk 6]
```diff
@@ -39,17 +39,15 @@ func (h *CSVHandler) GetSchemas() ([]*Schema, error) {
 		if err != nil {
 			return nil, err
 		}
+		schema.Columns = make([]*Column, len(values))
 		for i := range values {
 			if i == h.ColumnRowIndex-1 {
-				columns := []Column{}
 				for j := range values[i] {
-					column := Column{
+					schema.Columns[i] = &Column{
```
Comment (Mitu217) [inline datasource/handler/csv.go]: Nice👍

File: datasource/type.go [hunk 7]
```diff
@@ -18,6 +19,8 @@ func (t Type) String() string {
 		return "sql"
 	case Spreadsheet:
 		return "spreadsheet"
+	case Spanner:
+		return "spanner"
```
Comment (Mitu217) [inline datasource/type.go]: please need add type...

https://github.com/Mitu217/tamate/blob/f26419fb1780165c07d650480396dee0502c7130/datasource/datasource.go#L52

Comment (castaneai) [reply datasource/type.go]: done!

**Pull request merged**